src/main/kotlin/blackjack/domain/BlackJack.kt

```diff
@@ -1,55 +1,59 @@
 package blackjack.domain
 
-class BlackJack(val players: List<Player>) {
-
-    private var nowPlayer = 0
-    private var playCount = 0
-    private var isEnd = false
+class BlackJack(
+    val players: List<Player>,
+    val dealer: Dealer = Dealer(),
+    private val gameCards: GameCards = GameCards(),
+) {
 
     fun start() {
-        for (i in 0 until START_CARD_COUNT) players.forEach { it.draw() }
+        distributeInitialCard()
     }
 
-    fun isEnd(): Boolean {
-        if (!isEnd) {
-            checkAndChangeGamePlayer()
+    private fun distributeInitialCard() {
+        for (i in 0 until START_CARD_COUNT) {
+            dealer.addCard(gameCards.draw())
+            players.forEach { it.addCard(gameCards.draw()) }
         }
```

Comment:
2번 반복하면서 패를 뽑기보다는, 아래와 같이 `gameCards` 초기패 갯수만큼 카드를 뽑아주고, `dealer`, `player`는 복수 개의 card를 받을 수 있도록 변경해볼 수 있지 않을까요? 😃
```suggestion
        dealer.addCards(gameCards.drawInitCards())
        players.forEach { it.addCards(gameCards.drawInitCards()) }
```

Reply:
2장을 뽑는것은 블랙잭카드의 규칙이고, 블랙잭안에 해당 규칙이 존재해야 한다고 생각하여 포문을 제거하는 쪽으로 변경해보았습니다